src/App.tsx

```diff
@@ -1,32 +1,117 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoFilter } from './components/TodoFilter';
+
+enum SortType {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
 const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [loading, setLoading] = useState(false);
+  const [userId, setUserId] = useState<number>(0);
+  const [selectedTodo, setSelectedTodo] = useState<number>(0);
+  const [filteredBy, setFilteredBy] = useState<string>(SortType.ALL);
+  const [query, setQuery] = useState<string>('');
+  const [selectedUser, isUserSelected] = useState<boolean>(false);
```

Comment:
I meant when you have an initial value, like: 0, '', false -> you don't need to specify a type. Typescript will do it for you)